contribs/src/main/java/com/netflix/conductor/contribs/kafka/workers/ActiveWorkersMonitor.java

```diff
@@ -0,0 +1,338 @@
+        // ProductionExceptionHandler handles only exceptions on producer (exceptions occurring when sending messages
+        // via producer), it will not handle exceptions during processing of stream methods (mapValues(), branch(), etc.)
+        // You will need to wrap these methods in try / catch blocks.
+        // For consumer side, kafka streams automatically retry consuming record, because offset will not be changed
+        // until record is consumed and processed. Use setUncaughtExceptionHandler to log exception
+        // or send message to a failure topic.
+        builtStream.setUncaughtExceptionHandler((Thread thread, Throwable throwable) ->
+                // You can make it restart the stream, but you have to make sure that this thread is
+                // destroyed once a new thread is spawned to restart the kafka streams
+                logger.error(String.valueOf(throwable)));
+        return builtStream;
+    }
+
+    /**
+     * Builds and start the kafka stream topology and kafka stream object for receiving heartbeats from workers.
+     */
+    public void startWorkersStatusListenerStream() {
+        // Build the topology
+        Topology statusListenerTopology = buildStatusListenerTopology();
+        logger.debug("Status Listener Topology Description: {}", buildStatusListenerTopology().describe());
+        // Build/Create Kafka Streams object for receiving and processing workers status via kafka streams
+        KafkaStreams statusListenerStream = buildKafkaStream(statusListenerTopology);
+        logger.info("Starting Kafka Streams for receiving heartbeats from workers to Conductor");
+        statusListenerStream.start();
+        // Add shutdown hook to respond to SIGTERM and gracefully close the Streams application.
+        Runtime.getRuntime().addShutdownHook(new Thread(statusListenerStream::close));
+    }
+
+    /**
+     * Closes the inactive worker monitor timer thread
+     */
+    public void closeInActiveWorkerMonitor(){
+        timer.cancel();
+    }
+
+    /**
+     * A status object used to encapsulate the status of an active worker
+     */
+    private static class Status{
+
+        private final String taskName;
+        private long lastHeartbeat;
+        private long timeToNextHeartbeat;
+
+        public Status(final String taskName, final long workerInactive){
+            this.taskName = taskName;
+            this.lastHeartbeat = System.currentTimeMillis();
+            resetTimeToNextHeartbeat(workerInactive);
+        }
+
+        /**
+         * Sets thee latest heartbeat received and reset the inactive timer
+         * @param heartbeatMs The time of heartbeat
+         * @param workerInactiveTimer The specific length of time a heartbeat must be received from a worker/service to be considered active
+         */
+        public void updateLastHeartbeat(final long heartbeatMs, final long workerInactiveTimer){
+            setLastHeartbeat(heartbeatMs);
```

Comment:
Do not use setter and getters from within the class. Otherwise is basically obfuscation.